mlir/include/mlir/Dialect/Tosa/Transforms/TosaFoldCommon.h

```diff
@@ -13,33 +13,64 @@
 #define MLIR_DIALECT_TOSA_TRANSFORMS_TOSA_FOLD_COMMON_H
 
 #include <llvm/ADT/APFloat.h>
+#include <llvm/ADT/ArrayRef.h>
 #include <functional>
 #include <mlir/Dialect/Tosa/IR/TosaOps.h>
 #include <mlir/IR/PatternMatch.h>
 
 namespace mlir {
 namespace tosa {
 
-// Transform a tensor with the given transformation function.
+/// Type that represents tensor dimensions.
+using DimensionType = ArrayRef<int64_t>;
+
+/// Type for tensor offsets.
+using OffsetType = size_t;
+
+/// Transform a tensor with the given transformation function.
 DenseElementsAttr applyElementWise(
     const DenseElementsAttr &toTransform,
     const std::function<llvm::APFloat(const llvm::APFloat &, Type)> &toApply);
 
-/// Function that checks if arg is a dense TOSA constant float tensor
+/// Apply the given transformation function on the elements of the given
+/// tensors. If the input tensors do not match \p targetType, broadcasting is
+/// applied.
+DenseElementsAttr applyElementWise(
+    const DenseElementsAttr &, const DenseElementsAttr &, TensorType targetType,
```

Comment:
nit: I prefer having argument names in declarations to help readability

Reply:
Only is this particular case or everywhere? I tend to leave it out when there is no additional value, e.g., `PatternRewriter &rewriter`. I can adapt naming everything if this is more consistent with the rest.

Reply:
Ok, I can see leaving those out when they are obvious due to their unique type. Here, we have two arguments `DenseElementsAttr`. I don't think that I have seen argument names being elided in LLVM/MLIR, so it might come up during upstreaming again.